### PR TITLE
Adds system specs on orders and fulfillment reports

### PR DESCRIPTION
**Pull request by filipefurtad0**
#### What? Why?

Related to https://github.com/openfoodfoundation/openfoodnetwork/pull/9032.
Related to https://github.com/openfoodfoundation/openfoodnetwork/issues/7088.

Adds some tests to reports.

- before this PR `reports_spec.rb` only checked if the `Go` button rendered the page. This PR adds a stubbed order so some assertions can be made on the outcome of the reports.
- This PR moves some tests to it's own file (for orders and fulfillment). This report triggered some nasty bugs, so the idea is to go into some more detail, in future PRs, and leave `reports_spec.rb` for more superficial testing / complementary to some unit tests found on the `spec/lib/reports` folder.

#### What should we test?

Green build.

#### Release notes

Changelog Category:  Technical changes

The title of the pull request will be included in the release notes.

### CODE REVIEW

**Review by jibees — APPROVED** [review]
Great! 👍 
Maybe this is not needed to have in git history such a back and forth, and cleaning this history by squashing commits should be cleaner. What do you think?

File: spec/system/admin/reports_spec.rb [hunk 1]
```diff
@@ -108,20 +108,72 @@
...
+                                     I18n.t("report_header_shipping_cost"),
+                                     I18n.t("report_header_payment_method"),
+                                     I18n.t("report_header_distributor"),
+                                     I18n.t("report_header_distributor_address"),
+                                     I18n.t("report_header_distributor_city"),
+                                     I18n.t("report_header_distributor_postcode"),
+                                     I18n.t("report_header_shipping_method"),
+                                     I18n.t("report_header_shipping_instructions")]
+                                           .map(&:upcase)
+                                  ])
```
Comment (mkllnk) [inline spec/system/admin/reports_spec.rb:147]: In my opinion it's better to use the rendered strings like `"SKU"` in specs. It's easier to read and less code. It also means that it becomes more obvious if we made a mistake in a translation. And it's not like we change these strings all the time. We can do that in Transifex without changing en.yml and without affecting these specs.

File: spec/system/admin/reports/orders_and_fulfillment_spec.rb [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        expect(table).to eq([
+                              [I18n.t("report_header_hub"),
+                               I18n.t("report_header_customer"),
+                               I18n.t("report_header_email"),
+                               I18n.t("report_header_phone"),
+                               I18n.t("report_header_producer"),
+                               I18n.t("report_header_product"),
+                               I18n.t("report_header_variant"),
+                               I18n.t("report_header_quantity"),
+                               I18n.t("report_header_item_price", currency: currency_symbol),
```
Comment (mkllnk) [inline spec/system/admin/reports/orders_and_fulfillment_spec.rb:66]: Oh, and by testing against the rendered text, we also make sure that variables like the currency have been passed on correctly. It's easy to copy and paste incorrect code.

**Review by mkllnk — APPROVED** [review]
Thank you!

I have an opinion below but I don't think it's worth the additional effort to change it now. And maybe it's just my opinion?

Merging...

File: spec/system/admin/reports_spec.rb (hunk 1 above)
Comment (filipefurtad0) [reply spec/system/admin/reports_spec.rb:147]: Right, thank you for pointing this out - makes total sense, and I think I get the full length of [one of your previous comments](https://github.com/openfoodfoundation/openfoodnetwork/pull/8319#discussion_r725804103) now.

This file is to be updated soon (WIP) to add more tests on this report. I'll commit a change to replace the I18n assertions by simple strings on a separate PR :+1:

**Pull request merged**